util.go

```diff
@@ -172,3 +174,26 @@ func isFloat(k reflect.Kind) bool {
 		return false
 	}
 }
+
+func accessField(structVal reflect.Value, fieldIdx int, opts *options) (string, reflect.Type, reflect.Value, *options, tagOptions, string, bool) {
```

Comment:
ooof, these are a many return values. Didn't expect that. How about creating a struct named `fieldInfo`?

Reply:
Yeah I was thinking that it was becoming a lot of return values. I have updated it to now return a fieldInfo struct.